app/code/Magento/InventoryInStorePickup/Model/SearchCriteria.php

```diff
@@ -7,14 +7,18 @@
 
 namespace Magento\InventoryInStorePickup\Model;
 
-use Magento\InventoryInStorePickupApi\Api\Data\AddressInterface;
+use Magento\InventoryInStorePickupApi\Api\Data\SearchCriteriaInterface;
 
 /**
- * {@inheritdoc}
- * @codeCoverageIgnore
+ * @inheritdoc
  */
-class Address implements AddressInterface
+class SearchCriteria implements SearchCriteriaInterface
```

Comment:
Need to discuss.

**First Point.**
Implement \Magento\Framework\Api\SearchCriteriaInterface will lead to:
1) Add sort order (instead of distance sort order).
2) Add pager.
3) Add possibility to filter Sources by additionally specified fields/field groups.

Do we really need all of this?

**Second Point.**
Probably, we can make this class Immutable and pass all required data through constructor.

Reply:
Decided with first point - yes.

The only thing is confusing, Sources will be used for filtration by Search Criteria (e.g. **front_name** of Source instead of **name** of Pickup Location).